tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/NodesSecondaryChildrenRelatedTests.java

```diff
@@ -0,0 +1,285 @@
+                .createContent(new FileModel(filename, FileType.TEXT_PLAIN, "content"));
+        }
+
+        private static String generateRandomFolderNameWith(String folderSuffix)
+        {
+            return "folder" + folderSuffix + "_" + UUID.randomUUID();
+        }
+
+        private static String generateRandomFileName()
+        {
+            return "TestFile" + UUID.randomUUID() + ".txt";
+        }
+
+        private static String getCmisLocation(Object pathMap, String name)
+        {
+            return Stream.concat(
+                    Stream.of(pathMap)
+                        .filter(Objects::nonNull)
+                        .filter(path -> path instanceof Map)
+                        .map(Map.class::cast)
+                        .map(path -> path.get("elements"))
+                        .filter(Objects::nonNull)
+                        .filter(elements -> elements instanceof List)
+                        .map(List.class::cast)
+                        .flatMap(elements -> (Stream<?>) elements.stream())
+                        .skip(1)
+                        .filter(element -> element instanceof Map)
+                        .map(Map.class::cast)
+                        .map(element -> element.get("name"))
+                        .filter(Objects::nonNull)
+                        .filter(elementName -> elementName instanceof String)
+                        .map(String.class::cast),
+                    Stream.of(name))
+                .collect(Collectors.joining("/", "/", "/"));
+        }
+    }
+
+    /** Helper {@link Map} containing all created folders and allowing basic operations like creating and deleting a folder. */
+    @SuppressWarnings("PMD.TestClassWithoutTestCases")
+    public class TestFolders extends HashMap<String, Folder> implements Folders
+    {
+        /** Creates a folder in site's Document Library. */
+        @Override
+        public Folder createFolder(String folderSuffix)
+        {
+            Folder createdFolder = new Folder().createNestedFolders((Folder) null, folderSuffix).get(folderSuffix);
+            this.put(folderSuffix, createdFolder);
+            return createdFolder;
+        }
+
+        /** Creates multiple nested folders in site's Document Library. */
+        @Override
+        public Map<String, Folder> createNestedFolders(String... folderSuffixes)
+        {
+            Map<String, Folder> createdFolders = new Folder().createNestedFolders(null, folderSuffixes);
+            this.putAll(createdFolders);
+            return createdFolders;
+        }
+
+        @Override
+        public void delete(Folder folder)
+        {
+            folder.delete();
+            this.remove(folder);
+        }
+    }
+
+    public interface Folders extends Map<String, Folder>
+    {
+        Folder createFolder(String folderSuffix);
+
+        Map<String, Folder> createNestedFolders(String... folderSuffixes);
+
+        void delete(Folder folder);
+    }
```

Comment:
I think most of the helpers in this class would be widely useful. Perhaps we could consider adding them to the TAS framework in com-repo instead? I don't think this needs to be a base class for tests does it?

Reply:
I agree, it's already crossed my mind too, but I also think that firstly this logic should be adapted for more general use - this could be done in a separate task